src/atd/yarn_v0.atd

```diff
@@ -0,0 +1,49 @@
+
+(**************************************************************************)
+(*  Copyright 2014, Sebastien Mondet <seb@mondet.org>                     *)
+(*                                                                        *)
+(*  Licensed under the Apache License, Version 2.0 (the "License");       *)
+(*  you may not use this file except in compliance with the License.      *)
+(*  You may obtain a copy of the License at                               *)
+(*                                                                        *)
+(*      http://www.apache.org/licenses/LICENSE-2.0                        *)
+(*                                                                        *)
+(*  Unless required by applicable law or agreed to in writing, software   *)
+(*  distributed under the License is distributed on an "AS IS" BASIS,     *)
+(*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or       *)
+(*  implied.  See the License for the specific language governing         *)
+(*  permissions and limitations under the License.                        *)
+(**************************************************************************)
+
+type time <ocaml from="Ketrew_gen_base_v0.Time"> = abstract
+type host <ocaml from="Ketrew_gen_base_v0.Host"> = abstract
+type path <ocaml from="Ketrew_gen_base_v0.Path"> = abstract
+type program <ocaml from="Ketrew_gen_base_v0.Program"> = abstract
+type monitored_script <ocaml from="Ketrew_gen_base_v0.Monitored_script"> = abstract
+
+type daemonized_run <ocaml from="Ketrew_gen_daemonize_v0.Running"> = abstract
+
+type distributed_shell_parameters = {
+  hadoop_bin: string;
+  distributed_shell_shell_jar: string;
+  container_memory: [ GB of int | Raw of string ];
+  timeout: [ Seconds of int | Raw of string ];
```

Comment:
This one I made it up :)

(It is an argument to `DistributedShell.class`, not Yarn directly)

I have no idea of what exactly it may expect right now, and it may change completely in the next patch-level release.
I just don't want to constrain the user there because of a use-case we could not think of in advance.